### PR TITLE
Add a `RELEASE.md` and add @gotjosh as a mantainer

### PR DESCRIPTION
**Pull request by gotjosh**
I'm happy to help keep prometheus/common in a good place.

### CODE REVIEW

**Review by jesusvazquez — APPROVED** [review]
LGTM

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+While `prometheus/common` does not have a formal release process. We strongly encourage you follow these steps:
+
+1. Scan the list of available issues / PRs and make sure that You attempt to merge any pull requests that appear to be ready or almost ready
+2. Notify the maintainers listed as part of [`MANTAINERS.md`](MAINTAINERS.md) that you're going to do a release.
+
+With those steps done, you can proceed to cut a release.
+
+## How to cut an individual release
+
+There is no automated process for cutting a release in `prometheus/common`. A manual release using GitHub's release feature via [this link](https://github.com/prometheus/prometheus/releases/new) is the best way to go. The tag name must be prefixed with a `v` e.g. `v0.53.0` and then you can use the "Generate release notes" button to generate the release note automagically ✨. No need to create a discussion or mark it a pre-release, please do mark it as the latest release if needed.
```
Comment (SuperQ) [inline RELEASE.md:14]: Can we add a short paragraph about following semver?

For example, patch (v0.0.1) releases should contain bugfixes only.

Comment (gotjosh) [reply RELEASE.md:14]: Sounds good! I have added it in https://github.com/prometheus/common/pull/644/commits/5da977581ec0e5b2bb1618df8a4704262b04151a. WDYT?

File: MAINTAINERS.md [hunk 2]
```diff
@@ -1 +1,2 @@
 * Julien Pivotto <roidelapluie@prometheus.io> @roidelapluie
+* Josue (Josh) Abreu <josue.abreu@gmail.com> @gotjosh
```
Comment (ArthurSens) [inline MAINTAINERS.md:2]:
```suggestion
* Josue (Josh) Abreu <josue.abreu@gmail.com> @gotjosh
* Arthur Sens <arthur@prometheus.io> @ArthurSens 
```

**Review by ArthurSens — APPROVED** [review]
LGTM

**Review by SuperQ — APPROVED** [review]

**Pull request merged**